core/schemas/rbac.py

```diff
@@ -0,0 +1,133 @@
+import logging
+import re
+from functools import wraps
+from typing import TYPE_CHECKING, Any, ClassVar, Literal
+
+from fastapi import HTTPException, Request, status
+from pydantic import computed_field
+
+from core import database_arango
+from core.config.config import yeti_config
+from core.schemas import roles
+from core.schemas.model import YetiAclModel
+
+logger = logging.getLogger(__name__)
+
+
+if TYPE_CHECKING:
+    from core.schemas import user
+
+RBAC_ENABLED = yeti_config.get("rbac", "enabled", default=False)
+
+
+class Group(YetiAclModel, database_arango.ArangoYetiConnector):
+    _collection_name: ClassVar[str] = "groups"
+    _root_type: Literal["rbacgroup"] = "rbacgroup"
+
+    name: str
+    description: str | None = None
+
+    @classmethod
+    def load(cls, object: dict) -> "Group":
+        return cls(**object)
+
+    @computed_field(return_type=Literal["rbacgroup"])
+    @property
+    def root_type(self):
+        return self._root_type
+
+
+def permission_on_target(permission: roles.Permission):
```

Comment:
Shouldn't this function be part of apiv2/rbac.py? I find it confusing to mix the schemas and http request decorator here.

Reply:
Yes fair, will move them.

Reply:
On second thought, I'll keep them as it is. I see the files under apiv2/ as really endpoints defining actions to take by requests, and this code is more related to business logic surrounding access etc.

Reply:
Let's go!